src/components/Settings/index.jsx

```diff
@@ -114,7 +118,26 @@ export class Settings extends Component {
         shouldDisplayViewOfferButton =
           instanceModel.shouldDisplayOffers(data) || hasAnOffer(data)
 
-        managerUrlPremiumLink = instanceModel.buildPremiumLink(data)
+        const hasSubscription = flag('settings.subscription')
+
+        if (hasSubscription) {
+          // Contrived but we don't have access to client hooks nor client itself
+          stack.get.capabilities().then(res => {
+            const webLink = generateWebLink({
+              cozyUrl: stack.get.cozyURL(),
```

Comment:
You can access the `cozyUrl` directly with an instance of `CozyClient` with `client.getStackClient().uri`

Reply:
I'm not sure to understand, I don't think we have access to that in cozy bar v7? only v8

Reply:
I think we can access it because we set a [CozyProvider](https://github.com/cozy/cozy-bar/blob/v7-stable/src/index.jsx#L88) into the `renderBar`. However, I don't know whether there's HOC to access the instance of CozyClient

Reply:
I had issues with the HOC because it uses hooks under the hood. So I drilled the prop instead, only one level down so it looks ok to me, what do you think?

Reply:
that seems like a good compromise to me :+1